### PR TITLE
Respect the feature gate when exposing SEV device

### PR DESCRIPTION
**Pull request by vasiliy-ul**
**What this PR does / why we need it**:

Having `/dev/sev` in the list of permanently available devices may impose
a security risk as it can be requested and accessed by an arbitrary pod.
Stop advertizing the device when the corresponding feature gate is
disabled.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by vasiliy-ul** [comment]
> Looks great. Would you mind adding a small e2e test which ensures that the device is not there if disabled?
> 
> Typically such a test would just do
> 
> ```
> tests.DisableFeatureGate(virtconfig.SEVGate)
> Eventually(mynode).ShouldNot(Have())|
> ```
> 
> The cleanup should re-enable the feature gate.

I was just in doubt about the test since there are no SEV-capable nodes with `/dev/sev` in CI and therefore the number of available devices will be always zero. But I think I can use a hack with running `mknod /dev/sev/ c <maj> <min>` on the node (with some arbitrary major/minor). That will create a fake device node which will be picked by the device manager. I will experiment a bit with that and see if that works for e2e test.

File: tests/launchsecurity/sev.go [hunk 1]
```diff
@@ -15,26 +22,79 @@ import (
...
+
+		BeforeEach(func() {
+			virtClient, err = kubecli.GetKubevirtClient()
+			Expect(err).ToNot(HaveOccurred())
+
+			nodeName = tests.NodeNameWithHandler()
+			Expect(nodeName).ToNot(BeEmpty())
+
+			// Create a fake SEV device
+			mknodCmd := []string{"mknod", "/dev/sev", "c", "10", "124"}
```
Comment (rmohr) [inline tests/launchsecurity/sev.go:42]: Can we first check, by running a command on virt-handler, if the node has the sev device already? If so, we would not create and not remove it after the test.

**Review by rmohr — COMMENTED** [review]
Nice idea with the fake device. One improvement request.

File: tests/launchsecurity/sev.go (hunk 1 above)
Comment (rmohr) [reply tests/launchsecurity/sev.go:42]: Something simple from virt-handler would do, like `ls /proc/1/root/dev/sev`.

Comment (vasiliy-ul) [reply tests/launchsecurity/sev.go]: Agree, the check makes sense. Addressed.

**Comment by vasiliy-ul** [comment]
> { Failure tests/launchsecurity/sev.go:72
Tests which alter the global kubevirt configuration must not be executed in parallel
tests/testsuite/kubevirtresource.go:175}

Labeled the test as `[Serial]`.

**Comment by rmohr** [comment]
/lgtm
/approve

Great!

**Pull request merged**